src/App.tsx

```diff
@@ -15,71 +17,92 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+interface State {
+  selectedGood: string;
+}
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+export class App extends Component<{}, State> {
+  state = {
+    selectedGood: 'Jam',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  handleResetSelected = () => {
+    this.setState({ selectedGood: '' });
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  handleGoodsButton = (listItem: string) => {
+    const { selectedGood } = this.state;
+    const resetList = this.handleResetSelected;
+    const isSelected = listItem === selectedGood;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+    if (isSelected) {
+      resetList();
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+      return;
+    }
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+    this.setState({ selectedGood: listItem });
+  };
 
-        <tr data-cy="Good">
-          <td>
+  render(): ReactNode {
+    const { selectedGood } = this.state;
+    const goodsList = [...goods];
+
+    return (
+      <main className="section container">
+        <h1 className="title is-flex is-align-items-center">
+          {selectedGood
+            ? `${selectedGood} is selected`
+            : 'No goods selected'}
+
+          {selectedGood && (
             <button
-              data-cy="AddButton"
+              data-cy="ClearButton"
               type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+              onClick={this.handleResetSelected}
+              className="delete ml-3"
+            />
+          )}
+        </h1>
+
+        <table className="table">
+          <tbody>
+            {goodsList.map(item => {
+              const isSelected = item === selectedGood;
+              const trSelectedLight = classNames({
+                'has-background-success-light': isSelected,
+              });
+              const buttonSelectedBackground = classNames({
+                button: true,
+                'is-info': isSelected,
+              });
```

Comment:
you should do it where you assign attributes, no need to create a variable for that